src/simplify_dyn_ops.cpp

```diff
@@ -77,27 +139,51 @@ struct find_const_3in_slice
     {
         auto ins            = mr.result;
         auto inputs         = ins->inputs();
-        argument starts_arg = inputs.at(1)->eval();
-        argument ends_arg   = inputs.at(2)->eval();
-        if(not starts_arg.empty() and not ends_arg.empty())
+        auto slice_op       = any_cast<op::slice>(ins->get_operator());
+        auto set_attrs      = slice_op.get_set_attributes();
+        std::vector<int64_t> starts_vec;
+        std::vector<int64_t> ends_vec;
+        std::vector<int64_t> axes_vec;
+        if(set_attrs == slice_op.axes_only)
         {
-            std::vector<int64_t> starts_vec;
-            std::vector<int64_t> ends_vec;
-            starts_arg.visit([&](auto output) { starts_vec.assign(output.begin(), output.end()); });
-            ends_arg.visit([&](auto output) { ends_vec.assign(output.begin(), output.end()); });
-            auto slice_val = ins->get_operator().to_value();
-            auto axes_vec  = slice_val.at("axes").to_vector<int64_t>();
-            m.replace_instruction(
-                ins,
-                make_op("slice", {{"starts", starts_vec}, {"ends", ends_vec}, {"axes", axes_vec}}),
-                inputs.at(0));
+            // slice(data, starts, ends)
+            inputs.at(1)->eval().visit(
+                [&](auto output) { starts_vec.assign(output.begin(), output.end()); });
```

Comment:
Not directly related to this PR, but could we add a `to_vector` method to the `raw_data` class that would do this conversion:

```cpp
    template <class T>
    std::vector<T> to_vector() const
    {
        std::vector<T> result;
        visit([&](auto x) { result.assign(x.begin(), x.end()); });
        return result;
    }
```

Then you could just write `inputs.at(1)->eval().to_vector<int64_t>()`.

Reply:
If you do, comment it with Paul's usage example so other programmers will know how to use it.

Reply:
>  Do we want to make that enhancement?

If you want to make it in the PR you could or we can open another PR with the update.